istioctl/cmd/tag.go

```diff
@@ -0,0 +1,400 @@
+			}
+
+			return removeTag(context.Background(), client.Kube(), args[0], skipConfirmation, cmd.OutOrStdout())
+		},
+	}
+
+	return cmd
+}
+
+// setTag creates or modifies a revision tag.
+func setTag(ctx context.Context, kubeClient kubernetes.Interface, tag, revision string, w io.Writer) error {
+	revWebhooks, err := getWebhooksWithRevision(ctx, kubeClient, revision)
+	if err != nil {
+		return err
+	}
+	if len(revWebhooks) == 0 {
+		return fmt.Errorf("cannot find webhook with revision %q", revision)
+	}
+	if len(revWebhooks) > 1 {
+		return fmt.Errorf("found multiple canonical webhooks for revision %q", revision)
+	}
+
+	tagWebhook, err := buildTagWebhookFromCanonical(revWebhooks[0], tag, revision)
+	if err != nil {
+		return err
+	}
+
+	whs, err := getWebhooksWithTag(ctx, kubeClient, tag)
+	if err != nil {
+		return err
+	}
+
+	if len(whs) > 0 && !overwrite {
+		return fmt.Errorf("found an existing revision tag for %q, use the --overwrite flag to overwrite", tag)
+	}
+
+	// create or update webhook depending on whether it exists
+	// note there should be a single webhook for each revision tag
+	kubeWebhookClient := kubeClient.AdmissionregistrationV1().MutatingWebhookConfigurations()
+	if len(whs) == 0 {
+		_, err = kubeWebhookClient.Create(ctx, tagWebhook, metav1.CreateOptions{})
+	} else {
+		tagWebhook.ObjectMeta.ResourceVersion = whs[0].ObjectMeta.ResourceVersion
+		_, err = kubeWebhookClient.Update(ctx, tagWebhook, metav1.UpdateOptions{})
+	}
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(w, "Revision tag %s created\n", tag)
+	return nil
+}
+
+// removeTag removes an existing revision tag.
+func removeTag(ctx context.Context, kubeClient kubernetes.Interface, tag string, skipConfirmation bool, w io.Writer) error {
+	webhooks, err := getWebhooksWithTag(ctx, kubeClient, tag)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve tag with name %s: %v", tag, err)
+	}
+	if len(webhooks) == 0 {
+		return fmt.Errorf("revision tag %s does not exist", tag)
+	}
+
+	taggedNamespaces, err := getNamespacesWithTag(ctx, kubeClient, tag)
+	if err != nil {
+		return fmt.Errorf("could not retrieve namespaces dependent on tag: %s", tag)
+	}
+	// warn user if deleting a tag that still has namespaces pointed to it
+	if len(taggedNamespaces) > 0 && !skipConfirmation {
+		if !confirm(buildDeleteTagConfirmation(tag, taggedNamespaces), w) {
+			fmt.Fprintf(w, "Aborting operation.\n")
+			return nil
+		}
+	}
+
+	// proceed with webhook deletion
+	err = deleteTagWebhooks(ctx, kubeClient, webhooks)
+	if err != nil {
+		return err
+	}
+
+	fmt.Fprintf(w, "Revision tag %s removed\n", tag)
+	return nil
+}
+
+// listTags lists existing revision.
+func listTags(ctx context.Context, kubeClient kubernetes.Interface, writer io.Writer) error {
+	tagWebhooks, err := getTagWebhooks(ctx, kubeClient)
+	if err != nil {
+		return fmt.Errorf("failed to retrieve tags: %v", err)
+	}
+	if len(tagWebhooks) == 0 {
+		fmt.Fprintf(writer, "No tag webhooks found\n")
```

Comment:
Needs more context.  Something like "No Istio injection MutatingWebHookConfigurations to list".